plonky2/src/gates/gate_tree.rs

```diff
@@ -276,7 +276,7 @@ mod tests {
         assert!(
             gates_with_prefix
                 .iter()
-                .all(|(g, p)| g.0.degree() + g.0.num_constants() + p.len() <= 8),
+                .all(|(g, p)| g.0.degree() + g.0.num_constants() + p.len() <= 9),
```

Comment:
Replacing GMiMC with Poseidon changes the prefix lengths obtained by the Tree::traversal() greedy search earlier in this test. Also (probably related) the Poseidon gate has degree 7 while the GMiMC gate has degree 3. These combine to cause the value referred to above to go from 8 to 9. I think this is expected, but would appreciate confirmation from @dlubarov or @wborgeaud.

Reply:
Yes no problem with this, the gate tree that is generated in the recursive verifier also has degree 9.